addons/KOBUGE-games.CircularContainer/CircularContainer.gd

```diff
@@ -1,270 +1,302 @@
-		min_radius = max(min_radius, diagonal / 2)
-		max_radius += diagonal / 2
-		diagonals.push_back(diagonal)
-	
-	var key = str(diagonals)
-	if _cached_min_size_key == key:
-		return
-	
-#	var iter = 0
-	while max_radius > min_radius + 0.5:
-#		iter += 1
-		var new_radius = (max_radius + min_radius) / 2
-		
-		var angle_required = 0
-		for child in children:
-			angle_required += _get_max_angle_for_diagonal(_get_child_min_size(child).length(), new_radius)
-			
-		if angle_required < 2 * PI:
-			max_radius = new_radius # The angle needed is not high enough, we continue trying smaller values
-		else:
-			min_radius = new_radius # The angle needed is too high, we continue trying larger values
-	
-#	print(max_radius, "; found in ", iter, " iterations")
-	
-	_cached_min_size = Vector2(max_radius, max_radius) * 2 + min_child_size
-	_cached_min_size_key = key
-	
-	emit_signal("minimum_size_changed")
-
-func _default_animator(node, container_center, target_pos, time):
-	if node extends Control:
-		node.set_pos(container_center.linear_interpolate(target_pos - node.get_size() / 2 * time, time))
-	else:
-		node.set_pos(container_center.linear_interpolate(target_pos, time))
-	node.set_opacity(time)
-	if time == 0:
-		node.set_scale(Vector2(0.01,0.01))
-	else:
-		node.set_scale(Vector2(time,time))
-
-## Helpers ##
-
-func _get_filtered_children():
-	var children = get_children()
-	var i = children.size()
-	while i > 0:
-		i -= 1
-		var keep = false
-		if children[i] extends Control:
-			keep = true
-		elif _allow_node2d and children[i] extends Node2D:
-			keep = true
-		
-		if children[i] extends CanvasItem and children[i].is_hidden():
-			keep = false
-		
-		if !keep:
-			children.remove(i)
-	return children
-
-func _get_child_min_size(child):
-	if child extends Control:
-		var size = child.get_combined_minimum_size()
-		if _force_squares:
-			var s = max(size.x, size.y)
-			return Vector2(s,s)
-		return size
-	else:
-		return Vector2(0,0)
-
-func _get_child_stretch_ratio(child):
-	if child extends Control and (child.get_h_size_flags() & SIZE_EXPAND or child.get_h_size_flags() & SIZE_EXPAND):
-		return child.get_stretch_ratio()
-	elif child extends Node2D:
-		return 1
-	elif _force_expand:
-		return 1
-	else:
-		return 0
-
-func _get_max_angle_for_diagonal(diagonal, radius):
-	var fit_length = diagonal / 2
-	if fit_length > radius:
-		return PI
-	else:
-		return asin(fit_length / radius) * 2
-
-
-func _private(value = null):
-	print("Invalid access to private variable!")
+tool
+extends Container
+
+## Properties ##
+var _force_squares = false setget set_force_squares, _private
+var _force_expand = false setget set_force_expand, _private
+var _start_angle = 0 setget set_start_angle, get_start_angle
+var _percent_visible = 1 setget set_percent_visible, get_percent_visible
+var _appear_at_once = false
+var _allow_node2d = false setget set_allow_node2d, _private
+var _start_empty = false setget set_start_empty, _private
+var _custom_animator_func = null

```

Comment:
Why were the `setget`-s modified?

Reply:
It's showing an error if I keep it as _private

![image](https://user-images.githubusercontent.com/8458801/61955734-08d87400-af92-11e9-9e60-52e11658ca4b.png)

Reply:
That also happens if I add the setget after _appear_at_once var

Reply:
Maybe have `_private_get` and `_private_set`?

Reply:
Yep, that worked fine! Let me know if there is anything else I might have missed.

Also, thanks for the quick feedback!